sonar-db/src/main/resources/org/sonar/db/component/SnapshotMapper.xml

```diff
@@ -182,36 +158,29 @@
     order by created_at desc
   </select>
 
-
-  <update id="updateSnapshotAndChildrenLastFlagAndStatus" parameterType="map">
+  <update id="unsetIsLastFlagForComponentUuid" parameterType="map">
```

Comment:
in addition, one is avoiding doing the update if the flag is already set when the other isn't